src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          state[key] = actions[i].extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key in actions[i].keysToRemove) {
+          delete state[actions[i].keysToRemove[key]];
+        }
```

Comment:
[Checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) 1st item, don't use `for ... in` for array iteration